### PR TITLE
#4259 Correct unset on field array values

### PR DESCRIPTION
**Pull request by leapful**

### CODE REVIEW

File: src/useFieldArray.test.tsx [hunk 1]
```diff
@@ -4073,6 +4073,97 @@ describe('useFieldArray', () => {
       ).toHaveBeenCalledTimes(6);
     });
 
+    it('should unset field array values correctly on DOM removing', async () => {
```
Comment (bluebill1049) [inline src/useFieldArray.test.tsx:4076]: awesome! I will port your test case into v7 as well.

File: src/useFieldArray.test.tsx [hunk 2]
```diff
@@ -4073,6 +4073,97 @@ describe('useFieldArray', () => {
...
+        const { fields } = useFieldArray({
+          control,
+          name: `children[${childIndex}].nested`,
+        });
+
+        return (
+          <div>
+            {fields.map((field, index) => {
+              return (
+                <>
```
Comment (bluebill1049) [inline src/useFieldArray.test.tsx:4096]: minor: i think you are missing the `key` here.

File: src/useFieldArray.ts [hunk 3]
```diff
@@ -508,7 +508,7 @@ export const useFieldArray = <
       shouldUnregister && remove();
       resetFields();
       delete resetFunctions[name];
-      unset(fieldArrayValuesRef, name);
+      unset(fieldArrayValuesRef.current, name);
```
Comment (bluebill1049) [inline src/useFieldArray.ts:511]: 👯

**Review by bluebill1049 — COMMENTED** [review]
LGTM! just one minor comment.

File: src/useFieldArray.test.tsx (hunk 2 above)
Comment (leapful) [reply src/useFieldArray.test.tsx]: @bluebill1049 : Updated in the latest commit.

**Comment by leapful** [comment]
@bluebill1049 Cloud you please approve this pull request so we can merge it ?

**Comment by bluebill1049** [comment]
I will publish the patch tonight.

**Pull request merged**